client/packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/hooks/useStocktakeLineEdit.ts

```diff
@@ -1,31 +1,40 @@
-import { RecordPatch, ArrayUtils } from '@openmsupply-client/common';
-import { ItemRowFragment } from '@openmsupply-client/system';
 import {
-  StocktakeLineFragment,
-  UpsertStocktakeLinesMutation,
-  useStocktake,
-} from './../../../../api';
+  RecordPatch,
+  ArrayUtils,
+  useTranslation,
+  noOtherVariants,
+} from '@openmsupply-client/common';
+import { ItemRowFragment } from '@openmsupply-client/system';
+import { StocktakeLineFragment, useStocktake } from './../../../../api';
 import { DraftStocktakeLine, DraftLine } from '../utils';
 import { useNextItem } from './useNextItem';
 import { useDraftStocktakeLines } from './useDraftStocktakeLines';
+import { useStocktakeLineErrorContext } from 'packages/inventory/src/Stocktake/context';
 
 interface useStocktakeLineEditController {
   draftLines: DraftStocktakeLine[];
   update: (patch: RecordPatch<StocktakeLineFragment>) => void;
   addLine: () => void;
-  save: (lines: DraftStocktakeLine[]) => Promise<UpsertStocktakeLinesMutation>;
+  save: () => Promise<{ errorMessages?: string[] }>;
   isLoading: boolean;
   nextItem: ItemRowFragment | null;
-  isError: boolean;
 }
 
+export const errorMessage = (error: unknown): string => {
+  // Bugsnag it ?
+  if (error instanceof Error) return error.message;
+  return String(error);
+};
+
 export const useStocktakeLineEdit = (
   item: ItemRowFragment | null
 ): useStocktakeLineEditController => {
+  const t = useTranslation(['inventory']);
   const { id } = useStocktake.document.fields('id');
   const nextItem = useNextItem(item?.id);
   const [draftLines, setDraftLines] = useDraftStocktakeLines(item);
-  const { mutateAsync: save, isLoading, isError } = useStocktake.line.save();
+  const { mutateAsync, isLoading } = useStocktake.line.save();
+  const errors = useStocktakeLineErrorContext();
```

Comment:
bit of a :nit but the naming of the variable threw me.
`errors` would normally indicate an array of errors.. or some kind of collection at least.
This is some kind of errorContext or error Object, and there's functions on it which are called later on.

Reply:
`errorsContext` is much better thanks: 590d2924b5052b147580df86e7dc3c650a1ceb21